e2e/shortcut-action/shortcut-action.e2e-spec.ts

```diff
@@ -58,19 +58,39 @@ describe('ShortcutAction', function() {
-    let inputElemValue = inputElem.getAttribute('value');
-    inputElem.sendKeys(protractor.Key.chord(mod, protractor.Key.SHIFT, protractor.Key.UP));
-    let movedUpInputElem = page.getChildOfElementByCss(page.getElementById('accelerator_experiments.0.experiment'), 'textarea');
-    expect(movedUpInputElem.getAttribute('value')).toEqual(inputElemValue);
+    let currentFirstRow = page.getValuesOfChildrenById('keywords.0');
+    let currentSecondRow = page.getValuesOfChildrenById('keywords.1');
+    let currentElem = page.getChildOfElementByCss(page.getElementById('keywords.1.keyword'), 'textarea');
+    currentElem.sendKeys(protractor.Key.chord(mod, protractor.Key.SHIFT, protractor.Key.UP));
+    let targetFirstRow = page.getValuesOfChildrenById('keywords.0');
+    let targetSecondRow = page.getValuesOfChildrenById('keywords.1');
+    expect(targetFirstRow).toEqual(currentSecondRow);
+    expect(targetSecondRow).toEqual(currentFirstRow);
+    // Test for issue of overriding the new value with the old one when moving up because of triggering the commitValueChange()
+    // on blur with the old values. As a result the switched cells have the same values
+    expect(targetSecondRow).not.toEqual(targetFirstRow);
   });
 
   it(`should move row down using 'mod+shift+down' shortcut`, () => {
-    let inputElem = page.getChildOfElementByCss(page.getElementById('accelerator_experiments.0.experiment'), 'textarea');
-    let inputElemValue = inputElem.getAttribute('value');
-    inputElem.sendKeys(protractor.Key.chord(mod, protractor.Key.SHIFT, protractor.Key.DOWN));
-    let movedDownInputElem = page.getChildOfElementByCss(page.getElementById('accelerator_experiments.1.experiment'), 'textarea');
-    expect(movedDownInputElem.getAttribute('value')).toEqual(inputElemValue);
+   let currentRow = page.getValuesOfChildrenById('keywords.0');
+   let currentElem = page.getChildOfElementByCss(page.getElementById('keywords.0.keyword'), 'textarea');
+   currentElem.sendKeys(protractor.Key.chord(mod, protractor.Key.SHIFT, protractor.Key.DOWN));
+   currentElem = page.getChildOfElementByCss(page.getElementById('keywords.1.keyword'), 'textarea');
+   // Trigger move down shortcut two time in a row to test issue of not updating tabindexes correctly on sequential
+   // trigger of the shortcut. As a result the shortcut was not working properly.
+   currentElem.sendKeys(protractor.Key.chord(mod, protractor.Key.SHIFT, protractor.Key.DOWN));
+   let targetRow = page.getValuesOfChildrenById('keywords.2');
+   expect(targetRow).toEqual(currentRow);
+  });
+
+  it(`should update tabindexes correctly of the switched rows when moving up/down clicking the button`, () => {
+    let currentRowElem = page.getChildOfElementByCss(page.getElementById('accelerator_experiments.0.experiment'), 'textarea');
+    page.getChildOfElementByCss( page.getElementById('accelerator_experiments.1'), '.editor-btn-move-up').click();
+    let targetRowElem = page
```

Comment:
There is something quite confusing about this naming, your `currentRowElem` is the element 0 but then you take the action of click in element 1. (in the other tests `current` means the one you are going to do an action)

Then there is `targetRowElem` which is element 1. But if you click `up` on element 1 your target would be element 0 for better naming.

And then finally before the assertion, the comment says:

```
// moved element's tabindex must have increased by 1
```

but inside you check `targetTabIndex  - 1 == currentRowTabIndex`. Since you clicked the up button I would expect that that the `targetTabIndex  + 1 == currentRowTabIndex` if target was the element 0.